packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1304,41 +1304,46 @@ export function getSdkModelPropertyType(
 ): [SdkModelPropertyType, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
 
-  const clientParams = operation
-    ? context.__clientToParameters.get(getLocationOfOperation(operation))
-    : undefined;
-  const correspondingClientParams = clientParams?.find((x) =>
-    twoParamsEquivalent(context, x.__raw, type),
-  );
-  if (correspondingClientParams) return diagnostics.wrap(correspondingClientParams);
-  const base = diagnostics.pipe(getSdkModelPropertyTypeBase(context, type, operation));
+  let property = context.referencedPropertyMap?.get(type);
```

Comment:
Maybe we should initialize `context.referencePropertyMap` the first time we access it, instead of initializing it when we want to put the property in the cache in `updateReferencedPropertyMap()`?

Reply:
well I am mimicking the same way we did for types `referencedTypeMap` therefore I think this should be output the scope of this PR.
JS does not have as convenient mechanism as we have in dotnet that we could let a property to initialize the first time we use it. To do this we have to construct a `class`, but here we only have those interfaces, therefore I think what we have now is our best compromise